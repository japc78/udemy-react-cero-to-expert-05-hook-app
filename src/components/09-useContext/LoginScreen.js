import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginScreen = () => {
	// Tarea.
	// 1. Obtener la referencia al context
	// 2. Extrare el setUser
	const { setUser } = useContext(UserContext)

	// 3. Enviar el objeto con setUser
	//  {
	// 	id: 1234,
	// 	name: 'Bitcero',
	// 	email: 'dev@bitcero.com'
	// }

	const newUser =  {
		id: 1234,
		name: 'Bitcero',
		email: 'dev@bitcero.com'
	}

	const handleLogin = () => {
		setUser(newUser)
	}

	return (
		<div>
			<h1>LoginScreen</h1>
			<hr/>

			<button
				className="btn btn-primary"
				onClick={ handleLogin }
			>
					Login
			</button>
		</div>
	)
}
