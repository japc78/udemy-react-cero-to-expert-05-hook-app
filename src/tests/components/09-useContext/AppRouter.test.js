import React from 'react'
import '@testing-library/jest-dom'
import { mount } from 'enzyme'
import { UserContext } from '../../../components/09-useContext/UserContext'
import { AppRouter } from '../../../components/09-useContext/AppRouter'
import { user } from './fixtures/userData'

describe('Test about UseContext with <AppRouter /> ', () => {


	const wrapper = mount(
		<UserContext.Provider value = {{ user }}>
			<AppRouter />
		</UserContext.Provider>
	);

	test('should be shown correctly', () => {
		expect( wrapper ).toMatchSnapshot();
	});
})
