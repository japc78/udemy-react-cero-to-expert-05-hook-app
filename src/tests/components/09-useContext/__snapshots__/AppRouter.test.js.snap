// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test about UseContext with <AppRouter />  should be shown correctly 1`] = `
<div>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Navbar
    </a>
    <button
      aria-controls="navbarNavAltMarkup"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarNavAltMarkup"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarNavAltMarkup"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-link active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Home
        </a>
        <a
          aria-current={null}
          className="nav-link"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
        <a
          aria-current={null}
          className="nav-link"
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </div>
    </div>
  </nav>
  <div
    className="container"
  >
    <div>
      <h1>
        HomeScreen
      </h1>
      <hr />
      <pre>
        {
   "id": 1234,
   "name": "Bitcero",
   "email": "dev@bitcero.com"
}
      </pre>
    </div>
  </div>
</div>
`;
